packages/playwright/src/worker/testInfo.ts

```diff
@@ -476,7 +484,9 @@ export class TestInfoImpl implements TestInfo {
 
   snapshotPath(...pathSegments: string[]) {
     const legacyTemplate = '{snapshotDir}/{testFileDir}/{testFileName}-snapshots/{arg}{-projectName}{-snapshotSuffix}{ext}';
-    return this._resolveSnapshotPath(undefined, legacyTemplate, pathSegments);
+    // Assume a single path segment corresponds to `toHaveScreenshot(name)` so pass it as a string,
```

Comment:
What if it is `testInfo.snapshotPath('foo')`, don't we want to sanitize the only segment?

Reply:
Updated for better clarity.